src/AirPressureSensor.cc

```diff
@@ -19,6 +19,8 @@
 #include <ignition/common/Profiler.hh>
 #include <ignition/transport/Node.hh>
 
+#include <ignition/plugin/Register.hh>
```

Comment:
Is this include needed at every sensor? Shouldn't `SensorFactory.hh` be enough?

Reply:
According with the migration guide https://github.com/ignitionrobotics/ign-plugin/blob/94482af3e9b2cc1cbb7c2dbeb2c7846f8283f0f6/MIGRATION.md#registering-a-plugin

I added this header for registering plugins. All sensors are individual components that why I need to use `ignition/plugin/Register.hh` in the `cc` file. I tried to include this header (or `#include <ignition/plugin/RegisterMore.hh>`) in the `SensorFactory.hh` but I was not able to compile the code because an  undefined reference:

``/usr/bin/ld: ../lib/libignition-sensors4-imu.so.4.0.0~pre1: undefined reference `IgnitionPluginHook'
```

Reply:
> According with the migration guide

I see. I mean, we'd still be including that header, just indirectly. The way `ign-sensors` is exposing the plugin functionality is through `SensorFactory`. So technically each sensor doesn't need to know about `ign-plugins`, they just know they need to use the `IGN_SENSORS_REGISTER_SENSOR` macro from the `SensorFactory.hh` header.

> I tried to include this header (or #include <ignition/plugin/RegisterMore.hh>) in the SensorFactory.hh but I was not able to compile the code because an undefined reference:

`SensorFactory.hh` uses the `IGNITION_ADD_PLUGIN` macro from `ignition/plugin/Register.hh`, so I believe it should include that header. Otherwise, that macro won't be defined. And it probably needs to link against `ignition-plugin${IGN_PLUGIN_VER}::register` to fix the undefined reference. I believe that if you link to it publicly, then each individual sensor won't need to also link against `register`, only against `ign-sensors`'s core library.

Reply:
@chapulina, I was trying your suggestion but I'm not able to compile the code. I was having a look to other packages that make use of ignition-plugin and the approach it's similar to the one that I used here. For example:

  - Ign-gui plugins include the register header in the `.cc` files. Not in the core library. [Link1](https://github.com/ignitionrobotics/ign-gui/blob/master/src/plugins/world_control/WorldControl.cc#L21) [Link2](https://github.com/ignitionrobotics/ign-gui/blob/master/src/plugins/topic_echo/TopicEcho.cc#L20)
 - ign-gazebo. the same idea. [Link3](https://github.com/ignitionrobotics/ign-gazebo/blob/master/src/systems/contact/Contact.cc#L25) [Link4](https://github.com/ignitionrobotics/ign-gazebo/blob/de5b025968c9cf0cfbfd8a852458482e87c70c6c/src/gui/plugins/view_angle/ViewAngle.cc#L22)
 - ign-rendering: [Link5](https://github.com/ignitionrobotics/ign-rendering/blob/e22fd80e25aa89454a2be1d69891c83c7900f4c0/ogre/src/OgreRenderEngine.cc#L33) [Link6](https://github.com/ignitionrobotics/ign-rendering/blob/47ec9a398aa14a59cd4ef5bf2e1cdc7894e99f55/optix/src/OptixRenderEngine.cc#L20) [Link7](https://github.com/ignitionrobotics/ign-rendering/blob/e22fd80e25aa89454a2be1d69891c83c7900f4c0/ogre2/src/Ogre2RenderEngine.cc#L34)

Reply:
> it's similar to the one that I used here

I think the key difference here is that the `IGNITION_ADD_PLUGIN` macro is used in `SensorFactory`, not in each plugin. All the examples you gave use that macro directly on the files.

Does `SensorFactory.hh` work if you don't include `Register.hh` above it? If it doesn't, that sounds like an architecture flaw. A header should work when included by itself, without the need for extra includes above it, right?

Reply:
All plugin are using the [MACRO](https://github.com/ignitionrobotics/ign-sensors/blob/dd9759929d5dd50beb953ca23cafd0a0adf279ca/src/AirPressureSensor.cc#L213) but it's hidden under this other one [IGN_SENSORS_REGISTER_SENSOR](https://github.com/ignitionrobotics/ign-sensors/blob/766301b738d9a92a7b17fbdefdbc0f0487fb1bea/include/ignition/sensors/SensorFactory.hh#L184-L187)

Reply:
@chapulina  I can remove `IGN_SENSORS_REGISTER_SENSOR` and make use of `IGNITION_ADD_PLUGIN.

Reply:
>  I can remove IGN_SENSORS_REGISTER_SENSOR and make use of `IGNITION_ADD_PLUGIN.

That sounds like a big migration effort for users. But since our support for custom sensors is not really there yet (https://github.com/ignitionrobotics/ign-sensors/issues/9), maybe this isn't a big deal. I'd be ok with it. What do you think, @iche033 ?